### ISSUE TITLE
A great part of my experience with 'exercism' was having to use command line. #Independent #Confident #Play-around #Playground

### ISSUE THREAD

**Issue by Ornataweaver** [exercism/gui#60]
It lowered my fear of code. I think not having a gui would make a greater learning experience. It seems frightening to the students at first, but then they realize how enjoyable this is.
I say let's discuss this over before making a gui. It will also take a lot of developers and participants' time.
A programmer should learn to deal with shell one day or another, we can help them by making it easier for them, by telling them don't be afraid, it's very easy.
I remember my first best English teacher did the same and then the GIANT ENGLISH language became a hobby of mine. If not for that, maybe I wouldn't be able to be here today.

**Comment by antony-jr** [comment]
@Ornataweaver you are right but look at **CodeWars** , they have everything in the web , you don't have to touch the shell.  ( I like shell but I don't like it for everything , would you like browsing with shell ?)   

**You cannot read a book without the knowledge to read , right ?** , You first learn to read by simple task with a lot of help from others , then you go on doing it the hard way.   

We all started off with **Mac** or **Windows** and then onto **Linux** and **Shell**.

I really think the users must have a option, don't you think?

**The whole world knows that shell is where you can become the pro , one day they will get it!**
Before that we need to provide them with options and advice!

**There is a saying I like:**

> Any intelligent fool can make things bigger and more complex... It takes a touch of genius - and a lot   
> of courage to move in the opposite direction.
> --E. F. Schumacher

**Comment by antony-jr** [comment]
I think exercism should be easy as **1,2,3** , like **CodeWars**

**Comment by Ornataweaver** [comment]
You are right, in fact I want a discussion here to make solutions.
Suggestions lead to solutions:
Check https://try.github.io/levels/1/challenges/1
Github has made it easy to learn basics of 'git' and to push away the "shell fear" or "shellophobia". (I'm looking forward to make a word for it here, tell me if you find a nice word for it. #meme)
Yet not super-easy. I imagine an almost blank screen with a terminal window in it, like in the link. But not much text. Only:


### Type the command below then enter it into your machine:
```shell
git init
```
or a simpler sentence for the absolute beginner. these are sketches to work on.


Then on the second page you see:
### Congratulations, Wasn't that super-easy? You invoked your first command, You have learned the spine-chilling '$  shell'.
or a simpler sentence for the absolute beginner. these are sketches to work on.


Then we can make both a cli gui on the web and a graphical gui.
I say it again, one of the main reasons I really liked exercism was it's command line beginning. I don't know about other users, I suggest a survey if possible. We don't want to dispose our features when they are the reason of **our triumph**.

**Comment by Ornataweaver** [comment]
#### I'm not against transition, I say let's not make another codewars or codecademy or freeCodeCamp.

exercism v1 has the essence of coding in it. It's super simple, and it's focused on text. It's the first of it's kind.
But I'm not feeling it as I'm looking at https://v2.exercism.io/. I'm not against transition, I say let's not make another codewars or codecademy or freecodecamp. We have something here that none of them have. We should find the key features and strengthen them, not lose them.

Or maybe I'm just naturally resisting transition.

**Comment by Ornataweaver** [comment]
The features that I like:
- simplicity
- command line interface
- very important: I write the code on my computer. I'm not dependent on freecodecamp's in-site text editor. I right it here and I test it here. INDEPENDENT, Confident.
- very pragmatic: exercism does not explain too much from the very first task it gives you. You have to go out and search the web from the very beginning. Thus if you do the first step you will never give up. You are used to leaving and browsing around.
This was the problem freeCodeCamp had, and caused me to abandon it halfway. I grew dependent on it since it had every thing necessary to go on the first 100 hours. I was used to that green website and didn't like to search for the code I needed. So midway I was stuck.

I hope I have helped, since you guys definitely helped me in my life.

**Comment by Ornataweaver** [comment]
"Sublime Text" is a text editor with style; It doesn't have gui even for it's settings; All text;
Yet so popular.

**Comment by antony-jr** [comment]
Well I always use **VIM** but sometimes I do use **Kate**. I always like to have the options!
Anyways I'm not in the official team so bye. ❤️  I'm just a contributor on the **C and C++** tracks.
@Ornataweaver You need to open the issue in the exercism discussion repo , https://github.com/exercism/discussions/ to get connected with all others!

**Comment by antony-jr** [comment]
@Ornataweaver but I do recommend new users to get started with the **console** !

**Comment by Ornataweaver** [comment]
@antony-jr How can I move this to '/exercism/discussions/'?

**Comment by antony-jr** [comment]
@Ornataweaver simply open a [new issue](https://github.com/exercism/discussions/issues/new)

**Comment by Ornataweaver** [comment]
I should have put this discussion in https://github.com/exercism/v2-feedback/issues/1 instead of https://github.com/exercism/discussions/issues/208

**Comment by Ornataweaver** [comment]
And of course here. :D

**Comment by marnen** [comment]
IMHO (but I'm not on the core team) Exercism exists to teach programming and test-first development, not to help developers get friendly with the command line (except for the Bash track, I suppose :) ). All the tooling is optional, after all, although it certainly is useful. While I agree with @Ornataweaver that no developer should be afraid of the command line, I disagree that that _alone_ means that Exercism's tools should be CLI-only.

I personally don't think that GUI tooling would add much value to the way I personally use Exercism, although I'd certainly want to play with the present (or another) GUI and confirm that impression. I appreciate a well-designed UI in general, whether command-line or graphical.

**Comment by Ornataweaver** [comment]
@marnen Thank you for joining. I am not against gui, I say we should force users to do only one simple command in shell, then tell them "See how easy it is". Some tasks are easier with gui and some things are done better at command line.

**Comment by antony-jr** [comment]
@Ornataweaver I think I got your point , If I'm right , you want to implement a **interactive guide** on  **GUI** and the **website** which is similar to **GIT** tutorials by **github** , right?... (like [here](https://try.github.io/))

**Like this**
![Imgur](https://i.imgur.com/llrM804.png)

**Comment by antony-jr** [comment]
I can create something like that in **C++** and **Qt5 Framework** for desktops and **react** for a web-version ( **but** the site is in **ruby** but react can be embeded). (**So this will be Unofficial** or you must contact the author to do this but it won't be that easy!)

**EDIT**:
much better than **try.github.io**

**Comment by Ornataweaver** [comment]
@antony-jr You're almost getting my point. But you missed one key feature and that makes what you understood completely against what I'm trying to say.
That's #independent. I want the students to use their own "Terminal" and their own "text-editor" of choice. I don't want them to code in the website at all. At least not until they have learned how to be independent of coding on the site. After doing 5 or 10 exercises we can let them use embedded text-editor for editing only. Which might be extra work for the team here. I still don't recommend that extra work, there should be a discussion before spending so much time on embedded tex-editor.
I say this because of my experience inside freeCodeCamp. I didn't feel like a real programmer since I was dependent on the site to make only what was told to me. I couldn't play around. I didn't have a 'Playground'.

**Comment by antony-jr** [comment]
@Ornataweaver I think exercism gui is not actively developed , So I suppose we are stuck with terminal for good. :+1:

**Comment by Ornataweaver** [comment]
@antony-jr Thanks for the conversation.
I'm gonna build the 'IDEAs Platform', which will be a brain for the world to decide what to do before taking action. Yet I am building a center for interests and the paths I have in front of me. I am choosing my best Visions.
If you get that signal from sb or in yourself, contact me.
Also check FabLab movement.
